osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+             << ((versionInfo.dwFileVersionMS >> 0) & 0xffff) << "."
+             << ((versionInfo.dwFileVersionLS >> 16) & 0xffff) << "."
+             << ((versionInfo.dwFileVersionLS >> 0) & 0xffff);
+  r["version"] = versionStr.str();
+
+  return Status(0);
+}
+
+Status logCrashedModule(const MINIDUMP_MODULE_LIST* stream,
+                        unsigned char* const dumpBase,
+                        unsigned long long exAddr,
+                        Row& r) {
+  if (stream == nullptr)
+    return Status(1);
+
+  for (unsigned int i = 0; i < stream->NumberOfModules; i++) {
+    auto module = stream->Modules[i];
+    // Is the exception address within this module's memory space?
+    if ((module.BaseOfImage <= exAddr) &&
+        (exAddr <= (module.BaseOfImage + module.SizeOfImage))) {
+      auto modulePath =
+          reinterpret_cast<MINIDUMP_STRING*>(dumpBase + module.ModuleNameRva);
+      r["module"] = wstringToString(modulePath->Buffer);
+      return Status(0);
+    }
+  }
+  return Status(1);
+}
+
+// Pulls the memory at target address from the Minidump
+void* getMemAtTarget(const MINIDUMP_MEMORY_LIST* stream,
+                     unsigned long long target,
+                     unsigned char* const dumpBase) {
+  if (stream == nullptr)
+    return nullptr;
+
+  for (unsigned int i = 0; i < stream->NumberOfMemoryRanges; i++) {
+    auto memRange = stream->MemoryRanges[i];
+    if ((memRange.StartOfMemoryRange <= target) &&
+        (target < (memRange.StartOfMemoryRange + memRange.Memory.DataSize))) {
+      auto offset = target - memRange.StartOfMemoryRange;
+      return static_cast<void*>(dumpBase + memRange.Memory.Rva + offset);
+    }
+  }
+  return nullptr;
+}
+
+Status storeTEBAndPEB(const MINIDUMP_THREAD_LIST* threadStream,
+                      const MINIDUMP_MEMORY_LIST* memStream,
+                      unsigned char* const dumpBase,
+                      unsigned int tid,
+                      TEB*& teb,
+                      PEB*& peb) {
+  if ((threadStream == nullptr) || (memStream == nullptr))
+    return Status(1);
+
+  unsigned long long tebAddr = 0;
+  for (unsigned int i = 0; i < threadStream->NumberOfThreads; i++) {
+    auto thread = threadStream->Threads[i];
+    if (thread.ThreadId == tid) {
+      tebAddr = thread.Teb;
+    }
+  }
+  if (tebAddr == 0)
+    return Status(1);
+
+  auto result = getMemAtTarget(memStream, tebAddr, dumpBase);
+  if (result == nullptr)
+    return Status(1);
+  teb = static_cast<TEB*>(result);
+
+  auto pebAddr =
+      reinterpret_cast<unsigned long long>(teb->ProcessEnvironmentBlock);
+  result = getMemAtTarget(memStream, pebAddr, dumpBase);
+  if (result == nullptr)
+    return Status(1);
+  peb = static_cast<PEB*>(result);
+
+  return Status(0);
+}
+
+Status logBeingDebugged(const PEB* peb, Row& r) {
+  if (peb == nullptr)
+    return Status(1);
+
+  if (peb->BeingDebugged == 1) {
+    r["being_debugged"] = "true";
```

Comment:
I think you could avoid needing this function altogether by changing the column type here to an `INTEGER`, and then you could just use:
```
r["being_debugged"] = peb->BeingDebugged;
```
wherever it's called.